### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by JesterOrNot**
This is a template pull request please build off this!

### CODE REVIEW

**Comment by jzaefferer** [comment]
Needs a link to https://github.com/nushell/contributor-book eh?

**Comment by JesterOrNot** [comment]
> Needs a link to https://github.com/nushell/contributor-book eh?

Something like this?

*Note: for a more complete guide see https://github.com/nushell/contributor-book*

**Comment by JesterOrNot** [comment]
@jonathandturner any feedback?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Welcome to nushell!
+
+*Note: for a more complete guide see https://github.com/nushell/contributor-book*
+
+
+For speedy contributions open it in GitPod nu will be preinstalled with the latest build in a vscode like editor all from your browser
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/nushell/nushell)
+
+To get live support from the community see our [Discord](https://discordapp.com/invite/NtAbbGn), [Twitter](https://twitter.com/nu_shell) or File an issue or request more features here on [Github](https://github.com/nushell/nushell/issues/new/choose)!
```
Comment (JesterOrNot) [inline CONTRIBUTING.md]: I know the ordering of these things are a bit unorganized does anyone have an idea of how to structure it?

**Comment by jonathandturner** [comment]
Some topics we could hit:

* Collaboration

Like you suggest, we should point them at discord, and channels like #developing for code discussion and #design-discussion for discussion about the design of a new feature.

* How to suggest new features

We could talk about how you could come discuss new features on the discord or post a proposal to the issue tracker before spending time on writing a full PR. We're relaxed and happy to chat about features, but it's generally better to know what's coming :)

* Code style

We use (and require) using `rustfmt` in default settings. It's easy to install via `rustup component add rustfmt`.

* Where to get help

We can point them to the developer guide for information about the way Nu works under the surface, discord for chats so they can ask questions, and if people want help with directions in code, they can post a draft PR and ask for feedback.

**Comment by jonathandturner** [comment]
Anyone want to finish this? I think having a CONTRIBUTING.md would be a great addition

**Comment by notryanb** [comment]
In our discord developer channel, the following was mentioned: "we just landed https://github.com/nushell/nushell/pull/1142, which turns on clippy by default, so you may see some new warning with CI. To run it locally, install clippy with rustup component add clippy. Then you can do cargo clippy --all --features=stable"

This can be helpful information if added the "code style" section.

**Comment by JesterOrNot** [comment]
@notryanb 
should clippy be added to Gitpod? if its part of the workflow

**Comment by jonathandturner** [comment]
@notryanb @JesterOrNot - sorry for the slow responses, I didn't see these comments until just now.

How about we land this, and then you can send PRs to expand it with the new sections?

**Pull request merged**